### DISCUSSION TITLE
How to use system installed fonts just like Windows?

### DISCUSSION THREAD

**Discussion by thelastfantasy** [emilk/egui#1173] [Q&A]
And also, the menu currently opens when clicked, how can I change it to open when mousedown?

**Comment by emilk** [comment]
Using the fonts installed on the system should be doable if you know what folder they are in. See https://docs.rs/egui/latest/egui/struct.FontDefinitions.html for how to add your own fonts to egui.

**Comment by thelastfantasy** [comment]
Thank you.
You mean the `FontData::from_static` way? I use this to build with a CJK font (like Noto Sans) and the build size increased almost 20MB.

**Reply by emilk** [reply]
You can use `FontData::from_owned` and load the font from file or web instead.

**Comment by thelastfantasy** [comment]
```rust
let mut fonts = FontDefinitions::default();
fonts.font_data.insert(
    "NotoSerifCJK_SC".to_owned(),
    FontData::from_owned(include_bytes!("../fonts/NotoSerifCJKsc-Regular.otf").to_vec()),
);
```

Um, I tried this way to load a font but the built size is the same as ```FontData::from_static```, how can I fix it?

**Reply by emilk** [reply]
`include_bytes!` is what bakes the font data into the binary. You need to load it from disk with e.g. [`std::fs::read`](https://doc.rust-lang.org/std/fs/fn.read.html)

**Reply by thelastfantasy** [reply]
> `include_bytes!` is what bakes the font data into the binary. You need to load it from disk with e.g. [`std::fs::read`](https://doc.rust-lang.org/std/fs/fn.read.html)

Changed to std::fs::read and the build size decreased to 3MB! Thank you very much!


Also, just like the topic said, now the menu component can only hold clicks but mouseup or mousedown do not work, can you add these feature (event?) in the future?
